### DISCUSSION TITLE
Trouble saving Many-to-Many relationship field via Form

### DISCUSSION THREAD

**Discussion by MechaKaradi** [Budibase/budibase#4897] [Help] [answered]
Hi team, 
I am trying to build a small app to track classes and tuition taking place in our education camps. 

As a part of that, I need to build an attendance tracker for the students. I am using BB's internal databases for all of this. There are 3 tables; ```Teachers``` and ```Students``` are 2 independedent tables but that share 1 column for Teachers which has the same text (this was imported from a CSV so they are not relationships. The third table is ```Classes```  and contains a text and date column to identify the class as well as 1 Many to Many relationship column each for ```Teachers``` and ```Students```

What I am trying to do is as follows;

1. Make a form that creates a class. Within that form I would like to select 1 ```Teacher```
2. Based on the selcted teacher, filter the list of students ( there are over 300 students so filtering the list is required)
3. Select multiple students that have attended the class
4. Save the row.

I trired using the method suggested in #4579, and theat worked to create a multiple select filter that returns an array of '_id's from the student column. However, when passing taht  array to the ```Save Form``` action, nothing  gets saved in the DB.

Looking forward to any support that is possible.

Thanks a lot!
Karadi

**Comment by mjashanks** [comment, marked as answer]
Hi Karadi.

I believe I have created the scenario that you are asking for. My App export is below.

Key steps - In my "New Class" screen...
1. I added a `Data Provider` linked to "Teachers", with no filter
2. Inside, I added a `Table Block`, linked to `Students`
3. I added the following filter to the `Table Block`
`Teacher equals ... (Javascript function below)`

```javascript
const teacher = $("Teachers.Rows").find(r => r._id == $("Form.Fields.Teacher"));
return teacher.Name;
```
i.e. I am searching the `Teachers` data provider, for the teacher selected in the `Teacher` relationship picker, and returning the `Name`.

4. In my save button, I am manually saving the `Students` field using the selected rows of the table block. Note that I had to use the javascript `return $("Students table.Selected rows");`, rather than handlebars `{{ Students table.Selected rows }}` - as handlebars converts everything to a string!

[Budi School-export-1647267764430.txt](https://github.com/Budibase/budibase/files/8245639/Budi.School-export-1647267764430.txt)

**Reply by MechaKaradi** [reply]
Hey! There was a small error in the relationship between ```Students``` and ```Classes```` (it needs to be M2M) But other than that this really seems to solve my problem! Thank you so much for your help. I'm attaching the edited export here for anyone else that finds this useful. 

[UrbanSchool_v2-export-1647283538450.txt](https://github.com/Budibase/budibase/files/8247754/UrbanSchool_v2-export-1647283538450.txt)
